### ISSUE TITLE
.Net 8 Wasm vs Server Performance

### ISSUE THREAD

**Issue by CincySteve** [dotnet/runtime#90465]
My previous issue[ #45363](https://github.com/dotnet/runtime/issues/45363) about Wasm performance was closed when I had difficulty trying it in .Net 8 preview 7.  I've now created 2 new demo repositories, one for [Blazor Wasm](https://github.com/CincySteve/PerformanceDemoWasm) and one for [Blazor Server](https://github.com/CincySteve/PerformanceDemoServer).  They are showing that cpu-intensive calculations are taking roughly 22 times longer in Wasm vs Server using .Net 8 preview 7 via VS Community 2022 v17.7.0 without the debugger.  

[@lewing](https://github.com/lewing), does that difference  make sense after the Wasm performance improvements since .Net 6 when I first posted the issue?  Is there something I can do to dramatically reduce that difference?  Thanks.

**Comment by lewing** [comment]
Is this with or without AOT?

**Comment by CincySteve** [comment]
Thanks for responding on a Saturday.  Above and beyond. 

I've published the demo to Azure using AOT (that took some learning).   The performance difference is now close to only 2.5x instead of 22x.  That's good enough I think to prompt me to reconstruct the Wasm version of my app.  I can't get the original one to startup under .Net 7 or 8, though it does with 6.  I'm getting "Uncaught Error Error: Content hash is required
    at loadResourceWithCaching (localhost꞉5001/_framework/blazor.webassembly.js:1:49246)".

Anyway, thanks again for the help, on Saturday.

**Comment by kg** [comment]
> "Uncaught Error Error: Content hash is required
at loadResourceWithCaching (localhost꞉5001/_framework/blazor.webassembly.js:1:49246)".

This sounds like a bug, can you open an issue about it?

**Comment by CincySteve** [comment]
@kg I did something that made the Uncaught Error go away, but unfortunately am not sure just what because I made a number of changes at once.  I suspect it was updating the NuGet packages, which I discovered were calling for updates to .Net 8 versions.  Anyway, since the app now starts up, it clearly was something I did that triggered the obscure error.  A friendlier error would have helped, assuming that would be possible.
